src/encoded/types/cart.py

```diff
@@ -99,47 +88,46 @@ def get_or_create_cart_by_user(context, request):
     user = request.registry[COLLECTIONS]['user'].get(userid)
     if not user:
         raise HTTPBadRequest()
-    carts = _get_carts_by_user(request, userid, ['disabled', 'deleted '])
+    carts = get_cart_objects_by_user(request, userid, blocked_statuses)
+    return carts, user
+
+
+@view_config(context=Cart.Collection, request_method='GET', permission='save-carts', name='get-cart')
+def get_or_create_cart_by_user(context, request):
+    carts, user = _get_user_carts(request, ['disabled', 'deleted '])
     if not carts:
         cart = _create_cart(request, user)
+        cart_atids = None
+    else:
+        cart_atids = [cart['@id'] for cart in carts]
     request.response.status = 200
     return {
         'status': 'success',
         '@type': ['result'],
-        '@graph': carts or [cart]
+        '@graph': cart_atids or [cart]
     }
 
 
 @view_config(context=Cart.Collection, request_method='PUT', permission='save-carts', name='put-cart')
 def create_cart_by_user(context, request):
-    userid = [
-        p.replace('userid.', '')
-        for p in request.effective_principals
-        if p.startswith('userid.')
-    ]
-    if not userid:
-        raise HTTPBadRequest()
-    else:
-        userid = userid[0]
-    user = request.registry[COLLECTIONS]['user'].get(userid)
-    if not user:
-        raise HTTPBadRequest()
-    cart_name = None
-    user_carts = get_cart_objects_by_user(request, userid)
-    visible_user_carts = filter_carts_to_visible(request, user_carts)
-    countable_carts = [
-        cart
-        for cart in visible_user_carts
-        if cart['status'] != 'disabled'
-    ]
-    cart_atids = [cart['@id'] for cart in visible_user_carts] if visible_user_carts else []
+    user_carts, user = _get_user_carts(request, ['deleted'] if not 'group.admin' in request.effective_principals else [])
```

Comment:
Again inline if/else in a function call.

Reply:
Also they are called `carts` in GET and `user_carts` in PUT. Would be nice to make them consistent.

Reply:
Yeah, changed.